javascript/0523-continuous-subarray-sum.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * https://leetcode.com/problems/continuous-subarray-sum/
+ * Hasing
+ * Time O(n) | Space O(n)
+ * @param {number[]} nums
+ * @param {number} k
+ * @return {boolean}
+ */
+var checkSubarraySum = function(arr, k) {
+    let sum = 0;
+    const remainderMap = new Map([ [0, -1] ]);
+    
+    for(let i = 0; i < arr.length;  i++) {
+        sum += arr[i];
+        if(remainderMap.has(sum%k)) {
+            if(i - remainderMap.get(sum%k) > 1) {
+                return true;
+            } 
+        } else {
```

Comment:
Suggestion: Avoid else

Reply:
I don't know how it would be possible without the else case. But I have removed it and added an if stetment. I also removed the nested if condition. Please have a look. Let me know if I can do better :) Btw, I update the submission link in the initial comment. So we know the newly updated code is correct and submitting. If I forget to update the submission link please let me know.